### DISCUSSION TITLE
Best practice for connecting from the browser? (re-use cookie to connect ok?)

### DISCUSSION THREAD

**Discussion by garlandcrow** [pocketbase/pocketbase#2051] [Q&A] [answered]
I wanted to use the real time subscription feature so I assume that I can just use the JS SDK to connect directly from the browser side to the PB server. I setup everything based on documentation so when a user logs in, there is auth info saved in the cookie to `pb_auth` and I can query PB on the backend. To connect from the browser however, what is the best practice? 

Is it OK to create do this in the browser?
```ts
const pb = new Pocketbase(PUBLIC_DB_URL)
pb.authStore.loadFromCookie(cookieString, 'pb_auth')
```
Where `cookieString` comes from reading the cookie from in JS and passing it like this? I couldn't tell if all the examples are meant to only be run server side or how it should be handled from client side. I guess I could wrap everything in an API and keep it on the backend except for the realtime part.

**Comment by ganigeorgiev** [comment, marked as answer]
If your application was entirely client-side, you can directly authenticate from the browser without the need to load an external cookie.

But based on the description, it seems that you have an additional backend (eg. node) that act as a middleware between the client and PocketBase. If that's the case, then Yes, it is OK to reuse the cookie on the client-side.

_Side-note: if you want the browser client to be able to also modify the auth state (eg. you have a "logout" action on the client-side), you'll need to update `document.cookie` on `pb.authStore` change:_

```js
pb.authStore.onChange((token, model) => {
    document.cookie = pb.authStore.exportToCookie({ httpOnly: false });
});
```
